packages/core/src/provider/base.ts

```diff
@@ -461,6 +468,36 @@ export class SmartAccountProvider<
     };
   };
 
+  private applyUserOperationOverrides = async (
+    struct: Deferrable<UserOperationStruct>,
+    overrides?: UserOperationOverrides
+  ): Promise<Deferrable<UserOperationStruct>> => {
+    const _struct = await resolveProperties(struct);
+
+    struct.maxPriorityFeePerGas = applyOverride(
+      _struct.maxPriorityFeePerGas,
+      overrides?.maxPriorityFeePerGas
+    );
+    struct.maxFeePerGas = applyOverride(
+      _struct.maxFeePerGas,
+      overrides?.maxFeePerGas
+    );
+    struct.callGasLimit = applyOverride(
+      _struct.callGasLimit,
+      overrides?.callGasLimit
+    );
+    struct.verificationGasLimit = applyOverride(
+      _struct.verificationGasLimit,
+      overrides?.verificationGasLimit
+    );
+    struct.preVerificationGas = applyOverride(
+      _struct.preVerificationGas,
+      overrides?.preVerificationGas
+    );
```

Comment:
any way we can make this more future proof? maybe using `Object.entries().forEach` or something?

otherwise each time a new override is added we have to make sure to deal with it here too

Reply:
sure. will make the change. That would be okay because the user operation overrides are subset fields of user operations.